modules/account-lib/src/coin/stx/contractBuilder.ts

```diff
@@ -0,0 +1,139 @@
+import { BaseCoin as CoinConfig } from '@bitgo/statics';
+import BigNum from 'bn.js';
+import {
+  makeUnsignedContractCall,
+  UnsignedContractCallOptions,
+  PayloadType,
+  ContractCallOptions,
+  UnsignedMultiSigContractCallOptions,
+  ClarityValue,
+  parseToCV,
+  ClarityAbiType,
+} from '@stacks/transactions';
+import { TransactionType } from '../baseCoin';
+import { InvalidParameterValueError, InvalidTransactionError, BuildTransactionError } from '../baseCoin/errors';
+import { Transaction } from './transaction';
+import { TransactionBuilder } from './transactionBuilder';
+import { isValidAddress } from './utils';
+import { ClarityValueJson } from './iface';
+import { Utils } from '.';
+
+export class ContractBuilder extends TransactionBuilder {
+  private _options: UnsignedContractCallOptions | UnsignedMultiSigContractCallOptions;
+  private _contractAddress: string;
+  private _contractName: string;
+  private _functionName: string;
+  private _functionArgs: ClarityValue[];
+
+  constructor(_coinConfig: Readonly<CoinConfig>) {
+    super(_coinConfig);
+  }
+
+  initBuilder(tx: Transaction): void {
+    const txData = tx.toJson();
+    if (txData.payload === undefined) {
```

Comment:
Is there any other verification we can do on the presence/correctness of contractName, functionName, functionArgs?